cloud_experts_tutorials/rosa-mobb-aws-load-balancer-operator.adoc

```diff
@@ -0,0 +1,345 @@
++
+[source,terminal]
+----
+$ oc -n aws-load-balancer-operator get pods
+----
++
+You should see the following, if not wait a moment and retry:
++
+[source,terminal]
+----
+NAME                                                             READY   STATUS    RESTARTS   AGE
+aws-load-balancer-controller-cluster-6ddf658785-pdp5d            1/1     Running   0          99s
+aws-load-balancer-operator-controller-manager-577d9ffcb9-w6zqn   2/2     Running   0          2m4s
+----
+
+== Validate the deployment with Echo Server application
+
+. Deploy Echo Server Ingress with ALB: 
++
+[source,terminal]
+----
+$ oc apply -f https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/main/docs/examples/echoservice/echoserver-namespace.yaml
+$ oc adm policy add-scc-to-user anyuid system:serviceaccount:echoserver:default
+$ oc apply -f https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/main/docs/examples/echoservice/echoserver-deployment.yaml
+$ oc apply -f https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/main/docs/examples/echoservice/echoserver-service.yaml
+$ oc apply -f https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/main/docs/examples/echoservice/echoserver-ingress.yaml
+----
++
+. Curl the ALB ingress endpoint to verify the echoserver pod is accessible: 
++
+[source,termnial]
+----
+$ INGRESS=$(oc -n echoserver get ingress echoserver \
+    -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+  curl -sH "Host: echoserver.example.com" \
+    "http://${INGRESS}" | grep Hostname
+
+  Hostname: echoserver-7757d5ff4d-ftvf2
+----
++
+. Deploy Echo Server NLB Load Balancer:
++
+[source,terminal]
+----
+$ cat << EOF | oc apply -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: echoserver-nlb
+  namespace: echoserver
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-type: external
+    service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: instance
+    service.beta.kubernetes.io/aws-load-balancer-scheme: internet-facing
+spec:
+  ports:
+    - port: 80
+      targetPort: 8080
+      protocol: TCP
+  type: LoadBalancer
+  selector:
+    app: echoserver
+EOF
+----
++
+. Test the NLB endpoint:
++
+[source,terminal]
+----
+$ NLB=$(oc -n echoserver get service echoserver-nlb \
+  -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+ curl -s "http://${NLB}" | grep Hostname
+
+ Hostname: echoserver-7757d5ff4d-ftvf2
+----
+
+== Clean up
+
+. Delete the Operator and the AWS roles:
++
+[source,terminal]
+----
+$ oc delete subscription aws-load-balancer-operator -n aws-load-balancer-operator
+   aws iam detach-role-policy \
+     --role-name "${ROSA_CLUSTER_NAME}-alb-operator" \
+     --policy-arn $POLICY_ARN
+   aws iam delete-role \
+     --role-name "${ROSA_CLUSTER_NAME}-alb-operator"
+----
++
+. You can delete the policy:
++
+[source,terminal]
+----
+$ aws iam delete-policy --policy-arn $POLICY_ARN
+----
```

Comment:
@bmcelvee Here is the updated content with the `hello-openshift` application deployed and all content inline:

```
== Validate the deployment with a hello world application

. Create a new project: 
+
[source,terminal]
----
$ oc new-project hello-world
----
+
. Deploy a hello world application: 
+
[source,terminal]
----
$ oc new-app -n hello-world --image=docker.io/openshift/hello-openshift
----
+
. Configure a NodePort service for the ALB to connect to: 
+
[source,terminal]
----
$ cat << EOF | oc apply -f -
apiVersion: v1
kind: Service
metadata:
  name: hello-openshift-nodeport
  namespace: hello-world
spec:
  ports:
    - port: 80
      targetPort: 8080
      protocol: TCP
  type: NodePort
  selector:
    deployment: hello-openshift
EOF
----
+
. Deploy an ALB using the operator: 
+
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: hello-openshift-alb
  namespace: hello-world
  annotations:
    alb.ingress.kubernetes.io/scheme: internet-facing
spec:
  ingressClassName: alb
  rules:
    - http:
        paths:
          - path: /
            pathType: Exact
            backend:
              service:
                name: hello-openshift-nodeport
                port:
                  number: 80
EOF
----
+
. Curl the ALB ingress endpoint to verify the hello world application is accessible: 
+
[NOTE]
====
ALB provisioning takes a few minutes. If you receive an error that says `curl: (6) Could not resolve host:`, please wait and try again.  
====

[source,termnial]
----
$ INGRESS=$(oc -n hello-world get ingress hello-openshift-alb \
    -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
$ curl "http://${INGRESS}"
----
+
.Example output
[source,text]
----
Hello OpenShift!
----

. Next, deploy an NLB for your hello world application:
+
[source,terminal]
----
$ cat << EOF | oc apply -f -
apiVersion: v1
kind: Service
metadata:
  name: hello-openshift-nlb
  namespace: hello-world
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-type: external
    service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: instance
    service.beta.kubernetes.io/aws-load-balancer-scheme: internet-facing
spec:
  ports:
    - port: 80
      targetPort: 8080
      protocol: TCP
  type: LoadBalancer
  selector:
    deployment: hello-openshift
EOF
----
+
. Test the NLB endpoint:
+
[NOTE]
====
NLB provisioning takes a few minutes. If you receive an error that says `curl: (6) Could not resolve host:`, please wait and try again.  
====

[source,terminal]
----
$ NLB=$(oc -n hello-world get service hello-openshift-nlb \
  -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
$ curl "http://${NLB}"
----
+
.Example output
[source,text]
----
Hello OpenShift!
----

== Clean up

. Delete the hello world application namespace (and all the resources in the namespace):
+
[source,terminal]
----
$ oc delete ns hello-world
----
+
. Delete the Operator and the AWS roles:
+
[source,terminal]
----
$ oc delete subscription aws-load-balancer-operator -n aws-load-balancer-operator
$ aws iam detach-role-policy \
  --role-name "${ROSA_CLUSTER_NAME}-alb-operator" \
  --policy-arn $POLICY_ARN
$ aws iam delete-role \
  --role-name "${ROSA_CLUSTER_NAME}-alb-operator"
----
+
. You can delete the policy:
+
[source,terminal]
----
$ aws iam delete-policy --policy-arn $POLICY_ARN
----
```